src/resolvers/acl.js

```diff
@@ -37,5 +39,134 @@ export default {
 
       return dataSources.aclSourceDelete(args, context, parseResolveInfo(info))
     }
+  },
+
+  Acl: {
+    groups: async (source, args, context, info) => {
+      const { dataSources } = context
+      const resolvedInfo = parseResolveInfo(info)
+      const requestInfo = parseRequestedFields(resolvedInfo, {}, 'aclGroup')
+
+      const { groups } = source
+
+      const aclGroupResponse = camelcaseKeys(groups, { deep: true })
+
+      const groupIds = []
+
+      // Checks to see if there are groupId
+      Object.values(aclGroupResponse).forEach((value) => {
+        const { groupId, userType } = value
+
+        if (groupId && !userType) {
+          groupIds.push(groupId)
+        }
+      })
+
+      const params = {
+        params: {
+          limit: 1000,
+          name: ''
+        }
+      }
+
+      if (groupIds.length > 0) {
+        // Gets all groups from EDL
+        const result = await dataSources.groupSourceSearch(
+          params,
+          context,
+          requestInfo
+        )
+
+        const filteredData = result.items.filter((item) => groupIds.includes(item.id))
+
+        // Creates a map based on groupId
+        const edlResponseMap = new Map()
+        filteredData.forEach((item) => {
+          edlResponseMap.set(item.id, item)
+        })
+
+        // Combine the arrays based on matching groupId
+        const combinedGroupResponse = aclGroupResponse.map((item) => {
+          if (item.groupId && edlResponseMap.has(item.groupId)) {
+            return {
+              ...item,
+              ...edlResponseMap.get(item.groupId)
+            }
+          }
+
+          return item
+        })
+
+        return {
+          count: combinedGroupResponse.length,
+          items: combinedGroupResponse
+        }
+      }
+
```

Comment:
Add a comment about this condition, why would we get here and what would we expect the data to look like?